janis_core/tool/test_suite_runner.py

```diff
@@ -0,0 +1,217 @@
+import difflib
+import hashlib
+import os
+from typing import Dict, List, Any, Optional
+
+from janis_core.tool.tool import Tool
+from janis_core.tool.test_classes import TTestExpectedOutput, TTestCompared, TTestCase
+from janis_core.types import File, String, Array
+from janis_core.tool import test_helpers
+from janis_core.utils.secondary import apply_secondary_file_format_to_filename
+
+
+class ToolTestSuiteRunner:
+    """
+    A class to read tool test cases, run the test cases, and assert the expected output
+    """
+
+    def __init__(self, tool: Tool):
+        self.tool = tool
+
+    def run(self, input: Dict[str, str], engine):
+        """
+        Run a tool or workflow given a list of input using the specified engine
+        """
+        # Note, we do not want janis_core to depend on janis_assistant except for when we run these tests
+        # So, we only import this package here in this function
+        # Make sure the correct version of janis_assistant is installed first
+
+        test_helpers.verify_janis_assistant_installed()
+
+        from janis_assistant.main import run_with_outputs
+        from janis_assistant.management.configuration import JanisConfiguration
+
+        output_dir = os.path.join(os.getcwd(), "tests_output", self.tool.id())
+        config = JanisConfiguration(engine=engine)
+        output = run_with_outputs(
+            tool=self.tool, inputs=input, output_dir=output_dir, config=config
```

Comment:
Michael will add an `engine` input parameter here, so i don't need to set the JanisConfiguration.